docs/pages/machine-id/guides/github-actions.mdx

```diff
@@ -0,0 +1,206 @@
+CI/CD platform is needed.
+
+<Admonition type="note">
+Currently GitHub Actions with Machine ID are unsupported on GitHub Enterprise
+or self-hosted runners.
+</Admonition>
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A node that is a part of the Teleport cluster with [Server Access](https://goteleport.com/docs/server-access/introduction/).
+- A local workstation with `tsh` access to the cluster.
+- A Github repository with Github actions enabled. This guide uses the example `gravitational/example`
+repo, however this value should be replaced with your own unique repo.
+
+## Step 1/3. Create a join Token for GitHub Actions
+
+Github Actions will need it's own join token in order to join the cluster.
+Using a text editor of your choice, create the configuration YAML file `tokenconfig.yaml` for the
+join token that resembles the following:
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: github-token
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  roles: [Bot]
+  join_method: github
+  bot_name: github-demo
+  github:
+    allow:
+      - repository: gravitational/example
+```
+
+Let's go over the token configuration YAML in more detail:
+
+- `metadata.name` defines the name of the token. Note that this value will need to be used in other parts of the configuration later.
+- `metadata.expires` defines the date that the join token will expire. This example is set
+to the year `3000`, however care should be taken in production in order to Ensure
+that the expiration date will not be given a value that is greater than what is needed.
+- `spec.bot_name` represents the name of the bot that will be configured with Machine ID.
+Note that this value will need to be used in other parts of the configuration later.
+- `metadata.expires` defines the date that the join token will expire. This example is set
+to the year `3000`, however care should be taken in production in order to Ensure
+that the expiration date will not be given a value that is greater than what is needed.
+- `spec.roles` defines the role for the bot user. The value of `[Bot]` is a pre-existing role
+that ensures that only bots are allowed to join the cluster using this token, reducing the scope of privileges.
+- `spec.join_method` defines the join method the token is applicable for. Since this guide
+only focuses on GitHub Actions, this value should always be equal to `github`.
+- `spec.github.allow` is used to set rules for what GitHub Actions will be able to
+authenticate by using the token. In this example, the `gravitational/example`
+repository is used, and this example repository should be replaced with your own repo.
+
+Most of the configuration settings used in the example configuration file `tokenconfig.yaml`
+do not have any variable settings, however the `allow` block of `spec.github.allow` has additional
+configuration options to more strictly define the scope of access granted to the token:
+
+| **Setting** | **Behavior** | **Example** |
```

Comment:
We should change the tense of the descriptions here I think e.g

"The repository where the action is running."
"The name of the environment the action is running in."
"The account that triggered the GitHub action run."

I think with the future current tense it sounds a bit like you are configuring GitHub Actions rather than configuring a rule for a Github Action that is running.

It's important we nail these descriptions here because I don't think a reference for them exists anywhere else. If you are unsure of one in particular, I will be happy to provide a fuller, clearer explanation of what that value is. 

These are the descriptions provided by GitHub themselves (note that we don't support all of these, but for the ones we do support, we are directly passing the value from the GitHub):
![image](https://user-images.githubusercontent.com/16336790/199352174-c7101ea9-0e85-49ad-aba3-b12a5b4d00a3.png)

Reply:
I've cleaned up this section and changed the tense per your recommendations. I've additionally added some links, which I feel are justified given that some of this terminology is not Teleport specific and tied to GitHub concepts. I believe my understanding of the behavior for the allow block options are correct, but I absolutely welcome your technically critical eye here, I'm open to any and all feedback you might have.